src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -913,12 +928,28 @@ bool DirectoryService::RunConsensusOnDSBlockWhenDSPrimary() {
   BlockHash prevHash;
 
   // Determine the DS PoW winners.
-  unsigned int numOfProposedDSMembers = ComputeDSBlockParameters(
-      sortedDSPoWSolns, sortedPoWSolns, powDSWinners, dsWinnerPoWs,
-      dsDifficulty, difficulty, blockNum, prevHash);
+  unsigned int numOfProposedDSMembers =
+      ComputeDSBlockParameters(sortedDSPoWSolns, powDSWinners, dsWinnerPoWs,
+                               dsDifficulty, difficulty, blockNum, prevHash);
 
   // Determine the losers from the performance.
-  DetermineByzantineNodes(numOfProposedDSMembers, removeDSNodePubkeys);
+  unsigned int numByzantine =
+      DetermineByzantineNodes(numOfProposedDSMembers, removeDSNodePubkeys);
+
+  // Sort and trim the PoW solutions.
+  auto sortedPoWSolns = SortPoWSoln(allPoWs, true, numByzantine);
+
+  // Remove the DS solutions from the PoW solutions.
```

Comment:
I move the removing of DS solutions from the PoW solutions here from `ComputeDSBlockParameters`, after the DSBlock is computed and the sorted PoW solutions are generated.